src/adapters/surrealDB/index.ts

```diff
@@ -0,0 +1,189 @@
+import { trainCase } from 'case-anything';
+import type { Pipeline } from '../../types/pipeline/base';
+import type { EnrichedBormSchema, PipelineOperation } from '../../types';
+import { enrichBQLQuery } from '../../pipeline/preprocess/query/enrichBQLQuery';
+import { cleanQueryRes } from './pipeline/postprocess/query/cleanQueryRes';
+import { pascal, dash } from 'radash';
+import { QueryPath } from '../../types/symbols';
+import { produce } from 'immer';
+import type {
+	SurrealDbResponse,
+	EnrichedBqlQuery,
+	EnrichedBqlQueryEntity,
+	EnrichedBqlQueryAttribute,
+} from './types/base';
+
+const getSubtype = (
+	schema: EnrichedBormSchema,
+	kind: 'entities' | 'relations',
+	thing: string,
+	result: Array<string> = [],
+): Array<string> => {
+	const subtypes = Object.values(schema[kind])
+		.filter((itemSchema) => itemSchema.extends === thing)
+		.map((itemSchema) => itemSchema.name);
+
+	if (subtypes.length === 0) {
+		return result;
+	}
+
+	for (const subtype of subtypes) {
+		result.push(subtype);
+		getSubtype(schema, kind, subtype, result);
+	}
+
+	return result;
+};
+
+const convertEntityId = (attr: EnrichedBqlQueryAttribute) => {
+	return attr['$path'] === 'id' ? `meta::id(${attr['$path']}) as id` : `${attr['$path']}`;
+};
+
+const handleCardinality = (schema: EnrichedBormSchema, query: EnrichedBqlQuery) => (obj: Record<string, unknown>) => {
+	const thingType = query.$thingType;
+
+	if (thingType === 'entity') {
+		return obj;
+	}
+
+	const entitySchema = schema['relations'][query.$thing];
+
+	return produce(obj, (payload) => {
```

Comment:
is there any radash fn doing this or simplifying it?

Reply:
No problem, I have transformed this function with reduce now.